pkg/bpf/bpf_kmesh_common.go

```diff
@@ -94,11 +97,17 @@ func (sc *BpfSockConn) loadKmeshSockConnObjects() (*ebpf.CollectionSpec, error)
 	)
 	opts.Maps.PinPath = sc.Info.MapPath
 
-	spec, err = bpf2go.LoadKmeshCgroupSock()
-
-	if err != nil || spec == nil {
+	if sc.Info.VerLT513 {
+		spec, err = bpf2go.LoadKmeshCgroupSockCompat()
+	} else {
+		spec, err = bpf2go.LoadKmeshCgroupSock()
+	}
+	if err != nil {
 		return nil, err
 	}
+	if spec == nil {
+		return nil, fmt.Errorf("error: loadKmeshSockConnObjects() spec is nil")
```

Comment:
nit: how can this happen when err == nil

Reply:
Fixed, as spec won't be nil if err is nil